src/modules/75-cf/hooks/__tests__/useEnvironmentSelectV2.test.tsx

```diff
@@ -6,46 +6,163 @@
+            {
+              identifier: 'abc',
+              name: 'abc'
+            },
+            {
+              identifier: 'xyz',
+              name: 'xyz'
+            }
+          ]
+        }
+      },
+      loading: false,
+      refetch: jest.fn()
+    } as any)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('It should send back proper component with no "All Environments" option', async () => {
+    renderComponent()
+
+    const envInput = screen.getByRole('textbox', { name: 'cf.shared.selectEnvironment' })
+
+    await waitFor(() => {
+      expect(envInput).toHaveValue('abc')
+    })
+
+    userEvent.click(envInput)
+
+    await waitFor(() => {
+      expect(screen.getByText('abc')).toBeInTheDocument()
+      expect(screen.getByText('xyz')).toBeInTheDocument()
+      expect(screen.queryByText('common.allEnvironments')).not.toBeInTheDocument()
+    })
+  })
+
+  test('It should call SETPREFERENCE', async () => {
+    renderComponent()
+
+    const envInput = screen.getByRole('textbox', { name: 'cf.shared.selectEnvironment' })
+
+    await waitFor(() => {
+      expect(envInput).toHaveValue('abc')
+    })
+
+    userEvent.click(envInput)
+
+    const options = screen.getAllByRole('listitem')
+
+    await waitFor(() => {
+      expect(options[0]).toHaveTextContent('abc')
+      expect(options[1]).toHaveTextContent('xyz')
+    })
+    userEvent.click(options[1])
+
+    await waitFor(() => {
+      expect(setPreference).toHaveBeenCalledWith('xyz')
+      expect(refetchAllEnvironmentsFlags).not.toHaveBeenCalled()
+    })
+  })
+
+  test('It should send back proper component including "All Environments" option', async () => {
+    renderComponent({ allowAllOption: true }, { defaultFeatureFlagValues: { FFM_6683_ALL_ENVIRONMENTS_FLAGS: true } })
+
+    userEvent.click(screen.getByRole('textbox', { name: 'cf.shared.selectEnvironment' }))
+
+    await waitFor(() => {
+      expect(screen.getByText('common.allEnvironments')).toBeInTheDocument()
+      expect(screen.getByText('abc')).toBeInTheDocument()
+      expect(screen.getByText('xyz')).toBeInTheDocument()
+    })
+  })
+
+  test('It should fetch flags data on click of "All Environments" option', async () => {
+    renderComponent({ allowAllOption: true }, { defaultFeatureFlagValues: { FFM_6683_ALL_ENVIRONMENTS_FLAGS: true } })
+
+    const envInput = screen.getByRole('textbox', { name: 'cf.shared.selectEnvironment' })
+
+    await waitFor(() => {
+      expect(envInput).toHaveValue('abc')
+      expect(refetchAllEnvironmentsFlags).toHaveBeenCalledTimes(0)
+      expect(setPreference).toHaveBeenCalledTimes(1)
+    })
+
+    userEvent.click(envInput)
+
+    const options = screen.getAllByRole('listitem')
+
+    await waitFor(() => {
+      expect(options[0]).toHaveTextContent('common.allEnvironments')
+      expect(options[1]).toHaveTextContent('abc')
+      expect(options[2]).toHaveTextContent('xyz')
+    })
+    userEvent.click(options[0])
```

Comment:
```suggestion

    userEvent.click(options[0])
```
a wee line break here to separate the assertions from the actions